src/uct/tcp/tcp_iface.c

```diff
@@ -615,6 +617,69 @@ static UCS_CLASS_DEFINE_NEW_FUNC(uct_tcp_iface_t, uct_iface_t, uct_md_h,
                                  uct_worker_h, const uct_iface_params_t*,
                                  const uct_iface_config_t*);
 
+/* Fetch information about available network devices through an ioctl.  */
+static ucs_status_t uct_tcp_query_devices_ioctl(uct_md_h md,
+                                                uct_tl_device_resource_t **devices_p,
+                                                unsigned *num_devices_p)
+{
+    int sock, err, i;
+    uct_tl_device_resource_t *devices, *tmp;
+    unsigned num_devices;
+    ucs_status_t status;
+    struct ifconf conf;
+    struct ifreq reqs[10];
```

Comment:
btw, why only `10`?

can we do something like:
```C
...
    struct ifconf conf;
    ...

    conf.ifc_len = 0;
    conf.ifc_req = NULL;

    status = ucs_socket_create(AF_INET, SOCK_STREAM, &sock);
    if (status != UCS_OK) {
        goto out;
    }

    err = ioctl(sock, SIOCGIFCONF, &conf);
    if (err < 0) {
        ucs_error("ioctl(SIOCGIFCONF) failed: %m");
        status = UCS_ERR_IO_ERROR;
        goto out;
    }

    conf.ifc_req = ucs_malloc(1, conf.ifc_len, "ifreq");
    if (conf.ifc_req == NULL) {
        ucs_error("memory alocation failed");
        status = UCS_ERR_NO_MEMORY;
        goto out;
    }

    err = ioctl(sock, SIOCGIFCONF, &conf);
    if (err < 0) {
        ucs_error("ioctl(SIOCGIFCONF) failed: %m");
        status = UCS_ERR_IO_ERROR;
        goto out;
    }

    devices     = NULL;
    num_devices = 0;
    for (i = 0; i < (conf.ifc_len / sizeof(struct ifreq)); i++) {
        ...
    }

    *num_devices_p = num_devices;
    *devices_p     = devices;
    status         = UCS_OK;

out:
    ucs_free(conf.ifc_req);
    if (sock >= 0) {
        close(sock);
    }
    return status;
```

Reply:
You're right, it's much better this way (I wasn't sure `SIOCGIFCONF` allowed this).

Thanks!